sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -0,0 +1,429 @@
+        public static bool operator <(ResourceIdentifier left, ResourceIdentifier right)
+        {
+            return left is null ? right is not null : left.CompareTo(right) < 0;
+        }
+
+        /// <summary>
+        /// Compares one <see cref="ResourceIdentifier"/> with another instance.
+        /// </summary>
+        /// <param name="left"> The object on the left side of the operator. </param>
+        /// <param name="right"> The object on the right side of the operator. </param>
+        /// <returns> True if the left object is less than or equal to the right. </returns>
+        public static bool operator <=(ResourceIdentifier left, ResourceIdentifier right)
+        {
+            return left is null || left.CompareTo(right) <= 0;
+        }
+
+        /// <summary>
+        /// Compares one <see cref="ResourceIdentifier"/> with another instance.
+        /// </summary>
+        /// <param name="left"> The object on the left side of the operator. </param>
+        /// <param name="right"> The object on the right side of the operator. </param>
+        /// <returns> True if the left object is greater than the right. </returns>
+        public static bool operator >(ResourceIdentifier left, ResourceIdentifier right)
+        {
+            return left is not null && left.CompareTo(right) > 0;
+        }
+
+        /// <summary>
+        /// Compares one <see cref="ResourceIdentifier"/> with another instance.
+        /// </summary>
+        /// <param name="left"> The object on the left side of the operator. </param>
+        /// <param name="right"> The object on the right side of the operator. </param>
+        /// <returns> True if the left object is greater than or equal to the right. </returns>
+        public static bool operator >=(ResourceIdentifier left, ResourceIdentifier right)
+        {
+            return left is null ? right is null : left.CompareTo(right) >= 0;
+        }
+
+        /// <summary>
+        /// Add a provider resource to an existing resource id.
+        /// </summary>
+        /// <param name="providerNamespace"> The provider namespace of the added resource. </param>
+        /// <param name="resourceType"> The simple type of the added resource, without slashes (/),
+        /// for example, 'virtualMachines'. </param>
+        /// <param name="resourceName"> The name of the resource.</param>
+        /// <returns> The combined resource id. </returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ResourceIdentifier AppendProviderResource(string providerNamespace, string resourceType, string resourceName)
+        {
+            ValidateProviderResourceParameters(providerNamespace, resourceType, resourceName);
+            return new ResourceIdentifier(this, providerNamespace, resourceType, resourceName);
+        }
+
+        /// <summary>
+        /// Add a provider resource to an existing resource id.
```

Comment:
Add a child resource?

Reply:
Defining the conceptual difference between provider and child resources would be helpful here - if there are docs on it, a link would be helpful.

Reply:
My current understanding is think of the provider portion as routing information for ARM.  All API calls go into a single service and then are forwarded to the service to handle the call using that provider piece.  When you add a virtualNetwork ARM needs to know who the provider is so you need 4 pieces in the ID
/provider/Microsoft.Network/virtualNetworks/myVnet
When you then add a subnet you don't need the provider information again since its owned / routed to the same place as virtualNetworks so it looks like this
/provider/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet